File: backend/src/main/java/com/isravel/workhub/intelligence/DailyAttendance.java
package com.isravel.workhub.intelligence;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "daily_attendance")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DailyAttendance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(name = "attendance_date", nullable = false)
    private LocalDate attendanceDate;

    @Column(name = "first_punch")
    private LocalDateTime firstPunch;

    @Column(name = "last_punch")
    private LocalDateTime lastPunch;

    @Column(name = "total_working_minutes")
    private Integer totalWorkingMinutes = 0;

    @Column(name = "break_duration_minutes")
    private Integer breakDurationMinutes = 0;

    @Column(name = "lunch_duration_minutes")
    private Integer lunchDurationMinutes = 0;

    @Column(name = "status")
    private String status = "ABSENT";

    @Column(name = "is_late")
    private Boolean isLate = false;

    @Column(name = "late_minutes")
    private Integer lateMinutes = 0;

    @Column(name = "is_early_departure")
    private Boolean isEarlyDeparture = false;

    @Column(name = "early_departure_minutes")
    private Integer earlyDepartureMinutes = 0;

    @Column(name = "overtime_minutes")
    private Integer overtimeMinutes = 0;

    @Column(name = "scheduled_work_minutes")
    private Integer scheduledWorkMinutes = 0;

    @Column(name = "work_schedule_id")
    private Integer workScheduleId;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: flutter_app/lib/models/daily_attendance.dart
class DailyAttendance {
  final int id;
  final int employeeId;
  final String attendanceDate;
  final String? firstPunch;
  final String? lastPunch;
  final int totalWorkingMinutes;
  final int breakDurationMinutes;
  final int lunchDurationMinutes;
  final int totalMinutes;
  final int workingMinutes;
  final int breakMinutes;
  final String status;
  final int overtimeMinutes;
  final int scheduledWorkMinutes;
  final int? workScheduleId;
  final bool isLate;
  final int lateMinutes;
  final bool isEarlyDeparture;
  final int earlyDepartureMinutes;

  DailyAttendance({
    required this.id,
    required this.employeeId,
    required this.attendanceDate,
    this.firstPunch,
    this.lastPunch,
    required this.totalWorkingMinutes,
    required this.breakDurationMinutes,
    required this.lunchDurationMinutes,
    required this.totalMinutes,
    required this.workingMinutes,
    required this.breakMinutes,
    required this.status,
    required this.overtimeMinutes,
    required this.scheduledWorkMinutes,
    this.workScheduleId,
    required this.isLate,
    required this.lateMinutes,
    required this.isEarlyDeparture,
    required this.earlyDepartureMinutes,
  });

  factory DailyAttendance.fromJson(Map<String, dynamic> json) {
    return DailyAttendance(
      id: json['id'],
      employeeId: json['employeeId'],
      attendanceDate: json['attendanceDate'],
      firstPunch: json['firstPunch'],
      lastPunch: json['lastPunch'],
      totalWorkingMinutes: json['totalWorkingMinutes'] ?? 0,
      breakDurationMinutes: json['breakDurationMinutes'] ?? 0,
      lunchDurationMinutes: json['lunchDurationMinutes'] ?? 0,
      totalMinutes: json['totalMinutes'] ?? 0,
      workingMinutes: json['workingMinutes'] ?? 0,
      breakMinutes: json['breakMinutes'] ?? 0,
      status: json['status']?.toString() ?? 'ABSENT',
      overtimeMinutes: json['overtimeMinutes'] ?? 0,
      scheduledWorkMinutes: json['scheduledWorkMinutes'] ?? 0,
      workScheduleId: json['workScheduleId'],
      isLate: json['isLate'] ?? false,
      lateMinutes: json['lateMinutes'] ?? 0,
      isEarlyDeparture: json['isEarlyDeparture'] ?? false,
      earlyDepartureMinutes: json['earlyDepartureMinutes'] ?? 0,
    );
  }
}
